test/langtools/tools/javac/processing/model/util/elements/TestElementsProgrammatic.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8341342
+ * @summary Test Elements methods programmatically
+ * @modules jdk.compiler
+ * @run junit TestElementsProgrammatic
+ */
+
+import com.sun.source.util.JavacTask;
+import java.net.URI;
+import java.util.List;
+import java.util.function.Consumer;
+import javax.tools.JavaCompiler;
+import javax.tools.JavaFileObject;
+import javax.tools.SimpleJavaFileObject;
+import javax.tools.ToolProvider;
+
+import org.junit.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Programmatically test workings of various methods of Elements.
+ */
+public class TestElementsProgrammatic {
+
+    private final JavaCompiler systemCompiler = ToolProvider.getSystemJavaCompiler();
+
+    @Test
+    public void automaticallyEnter() {
+        //make sure the get{All,}{Module,Package,Type}Element{s,} methods will automatically enter:
+        List<Consumer<JavacTask>> testCases = List.of(
```

Comment:
You can use `@ParameterizedTest @MethodSource("taskFactory")` for each test case, and `static JavacTask taskFactory() { /* ... */}` for junit to call to create the task argument for each test.